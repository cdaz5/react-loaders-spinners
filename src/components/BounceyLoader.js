import React from 'react';
import styled, { keyframes } from 'styled-components';
import { string, number, bool } from 'prop-types';

const bounce = keyframes`
  0%, 75%, 100% {
    transform: translateY(0px)
  } 
  25% {
    transform: translateY(-30px)
  }
`;

const BounceLoader = styled.div`
	display: ${props => (props.loading ? 'flex' : 'none')};
	align-items: center;

	span {
		margin: ${props => (props.spaceBetween ? `0 ${props.spaceBetween}px` : '0 10px')};
		display: block;
		width: ${props => (props.width ? `${props.width}px` : '20px')};
		height: ${props => (props.height ? `${props.height}px` : '20px')};
		border-radius: 50%;
		background: ${props => (props.pColor ? props.pColor : 'red')};

		&:nth-child(1) {
			animation: ${bounce} 1s ease-in-out infinite;
		}

		&:nth-child(2) {
			animation: ${bounce} 1s ease-in-out 0.33s infinite;
		}

		&:nth-child(3) {
			animation: ${bounce} 1s ease-in-out 0.66s infinite;
		}

		&:nth-child(4) {
			animation: ${bounce} 1s ease-in-out 0.66s infinite;
		}
	}
`;

const BounceyLoader = props => (
  <BounceLoader {...props}>
    <span />
    <span />
    <span />
  </BounceLoader>
);

BounceyLoader.propTypes = {
  width: number,
  height: number,
  loading: bool.isRequired,
  pColor: string,
  spaceBetween: number
};

BounceyLoader.defaultProps = {
  width: 20,
  height: 20,
  pColor: 'red',
  loading: true,
  spaceBetween: 10
};

export default BounceyLoader;
