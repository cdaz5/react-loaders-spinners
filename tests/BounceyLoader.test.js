import React from 'react';
import renderer from 'react-test-renderer';

import BounceyLoader from '../src/components/BounceyLoader';

describe('BounceyLoader', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<BounceyLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
