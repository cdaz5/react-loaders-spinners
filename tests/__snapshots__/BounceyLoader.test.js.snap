// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BounceyLoader matches snapshot 1`] = `
<div
  className="sc-bdVaJa ksVdhP"
  height={20}
  width={20}
>
  <span />
  <span />
  <span />
</div>
`;
